client/app/types/course/assessment/question/voice-responses.ts

```diff
@@ -0,0 +1,27 @@
+import { AvailableSkills, QuestionFormData } from '../questions';
+
+type OptionalIfNew<T extends 'new' | 'edit'> = T extends 'new'
+  ? undefined
+  : never;
+
+export interface VoiceData<T extends 'new' | 'edit' = 'edit'> {
+  question: QuestionFormData | OptionalIfNew<T>;
+}
+
+export interface VoiceFormData<T extends 'new' | 'edit' = 'edit'>
+  extends VoiceData<T>,
+    AvailableSkills {
+  allowRandomization: boolean;
```

Comment:
What is this?

Reply:
I'm really sorry. At first I thought it's the abstraction to allow randomization of skills (I know it sounds a bit stupid).. apparently after I re-read it again, it's specific for MCQ/MRQ and thus I have deleted it